Purchases/Public/RCPurchases.m

```diff
@@ -821,14 +824,16 @@ - (void)sendCachedPurchaserInfoIfAvailable {
 
 - (void)updateAllCachesIfNeeded {
     RCDebugLog(@"applicationDidBecomeActive");
-    if ([self.deviceCache isPurchaserInfoCacheStale]) {
-        RCDebugLog(@"PurchaserInfo cache is stale, updating caches");
-        [self fetchAndCachePurchaserInfoWithCompletion:nil];
-    }
-    if ([self.deviceCache isOfferingsCacheStale]) {
-        RCDebugLog(@"Offerings cache is stale, updating caches");
-        [self updateOfferingsCache:nil];
-    }
+    [self.systemInfo isApplicationBackgroundedWithCompletion:^(BOOL isAppBackgrounded) {
+        if ([self.deviceCache isPurchaserInfoCacheStaleForAppUserID:self.appUserID isAppBackgrounded:isAppBackgrounded]) {
+            RCDebugLog(@"PurchaserInfo cache is stale, updating caches");
+            [self fetchAndCachePurchaserInfoWithCompletion:nil isAppBackgrounded:isAppBackgrounded];
+        }
+        if ([self.deviceCache isOfferingsCacheStaleWithIsAppBackgrounded:isAppBackgrounded]) {
+            RCDebugLog(@"Offerings cache is stale, updating caches");
+            [self updateOfferingsCache:nil isAppBackgrounded:isAppBackgrounded];
+        }
+    }];
```

Comment:
would love to move all this stuff into a new class, but it feels too risky for this particular feature